pkg/infra/game/math/math.go

```diff
@@ -23,3 +31,53 @@ func CalculateMonsterDamage(n uint, hp uint, st uint, th float32, l uint, cl uin
 func GetNextLevelMonsterValues(gameConfig config.GameConfig, currentLevel uint) (uint, uint) {
 	return CalculateMonsterHealth(gameConfig.InitialNumAgents, gameConfig.Stamina, gameConfig.NumLevels, currentLevel+1), CalculateMonsterDamage(gameConfig.InitialNumAgents, gameConfig.StartingHealthPoints, gameConfig.Stamina, gameConfig.ThresholdPercentage, gameConfig.NumLevels, currentLevel+1)
 }
+
+func NumberPotionDropped(P float64, N uint) uint {
+	delta := CalculateDelta()
+	return uint(delta * P * float64(N))
+}
+
+func NumberEquipementDropped(E float64, N uint) uint {
+	delta := CalculateDelta()
+	return uint(delta * E * float64(N))
+}
+
+// function encapsulated to use same random val
+func GetPotionDistribution(N uint) (uint, uint) {
+	tau := rand.Float64()
+	// hardcoded by design
+	P := 0.2
+	NumberHealthPotionDropped := uint((tau) * float64(NumberPotionDropped(P, N)))
+	NumberStaminaPotionDropped := uint((1 - tau) * float64(NumberPotionDropped(P, N)))
+	return NumberHealthPotionDropped, NumberStaminaPotionDropped
+}
+
+// tau recalculated for equipement and potions
+func GetEquipmentDistribution(N uint) (uint, uint) {
+	tau := rand.Float64()
+	// hardcoded by design
+	E := 0.15
+	NumberWeaponDropped := uint((tau) * float64(NumberEquipementDropped(E, N)))
+	NumberShieldDropped := uint((1 - tau) * float64(NumberEquipementDropped(E, N)))
+	return NumberWeaponDropped, NumberShieldDropped
+}
+
+func GetWeaponDamage(X uint, N uint) uint {
+	delta := CalculateDelta()
+	return uint(math.Ceil((delta * float64(X)) / (4.0 * float64(N) * 0.8)))
+}
+
+func GetShieldProtection(Y uint, N uint) uint {
+	delta := CalculateDelta()
+	return uint(math.Ceil((delta * float64(Y) * 0.5) / (float64(N) * 0.8)))
+}
+
+func GetHealthPotionValue(Y uint, N uint) uint {
+	delta := CalculateDelta()
+	return uint(math.Ceil((delta * float64(Y) * 5.0) / (float64(N) * 0.8)))
+}
+
+func GetStaminaPotionValue(X uint, N uint) uint {
+	delta := CalculateDelta()
+	return uint(math.Ceil((delta * float64(X)) / (float64(N) * 0.8)))
+}
```

Comment:
These are useful but should probably open a new issue to make sure they're being used where needed in infra.

Other than that if the equations are correct then this lgtm. Maybe wait for @BusterBlack to double check. (Don't forget to rebase before merging or I can do it for you!)